src/download.cpp

```diff
@@ -206,6 +206,13 @@ void DownloadFile(const std::string& url, IDownloadSink *sink, int flags)
 
     if ( !filename_set )
     {
+        char ioptionurl[512];
+        DWORD ousize = 512;
```

Comment:
In the static buffer case, this should be `= sizeof(ioptionurl)` — it’s safer. Why ‘ousize’, BTW? Was that meant to be ‘outsize’?